### PR TITLE
Workflows UI and Bulk Actions

### LINKED ISSUE #1715: Improve Bulk Actions UI
- Only display Bulk Actions dropdown button when at least one item is selected.
- Add a "check all" checkbox.
- Apply Bulk Actions to list of Workflow instances too.

### PR DESCRIPTION
**Pull request by agriffard**
Fixes #1715

image

### CODE REVIEW

File: src/OrchardCore.Modules/OrchardCore.Workflows/Views/WorkflowType/Index.cshtml [hunk 1]
```diff
@@ -1,36 +1,61 @@
...
+                </div>
+                <div class="col-xl-4 mb-xl-n1">
+                    <span class="dropdown" id="actions" style="display:none">
+                        <button class="btn btn-light dropdown-toggle" type="button" id="bulk-action-menu-button" data-toggle="dropdown" aria-haspopup="true" aria-expanded="false">
+                            @T["Actions"]
+                        </button>
+                        <span class="dropdown-menu dropdown-menu-right" aria-labelledby="bulk-action-menu-button">
+                            @foreach (var item in Model.Options.WorkflowTypesBulkAction)
+                            {
+                                <a class="dropdown-item" href="javascript:void(0)" data-action="@item.Value" data-title="@T["Bulk Action"]" data-message="@T["Are you sure you want to {0} these items?", @item.Text.ToLower()]">@item.Text</a>
```
Comment (sebastienros) [inline src/OrchardCore.Modules/OrchardCore.Workflows/Views/WorkflowType/Index.cshtml]: This is bad usage of localization. You should not include a localized string in another one like this, at least not in this case, you can't assume that the "verb" will behave the same way in all languages. A better solution here would be to do a switch case on the action, and use different full strings

File: src/OrchardCore.Modules/OrchardCore.Workflows/Views/WorkflowType/Index.cshtml [hunk 2]
```diff
@@ -64,14 +87,70 @@
...
+        var itemsCheckboxes = $(":checkbox[name='itemIds']");
+
+        $(".filter-options input").on("change", function () {
+            $("[name='submit.Filter']").click();
+        });
+
+        $(".dropdown-menu .dropdown-item").on("click", function () {
+            if ($(":checkbox[name='itemIds']:checked").length > 1) {
+                var $this = $(this);
+                confirmDialog({title: $this.data('title'), message: $this.data('message'), callback: function(r) {
```
Comment (sebastienros) [inline src/OrchardCore.Modules/OrchardCore.Workflows/Views/WorkflowType/Index.cshtml]: @jpiquot added a new syntax. Can we still use this one or do we have to use the new one everywhere?

Comment (hishamco) [reply src/OrchardCore.Modules/OrchardCore.Workflows/Views/WorkflowType/Index.cshtml]: You mean @jptissot :)

Spread syntax

**Comment by agriffard** [comment]
@sfmskywalker Do you think it would be possible to find a way to filter the workflow instances on their Workflow Status?
We would need the corresponding ListAsync and CountAsync() methods.

**Comment by sfmskywalker** [comment]
@agriffard Most certainly. I’ll send a PR tomorrow.

**Comment by sfmskywalker** [comment]
I pushed some changes to your branch: https://github.com/OrchardCMS/OrchardCore/commit/7c44ba4991c794cdbc46f3fb531257bc04ce27c4

**Comment by agriffard** [comment]
Thank you @sfmskywalker 
Filter on Workflow status added.

The migration to add the Status column worked but every existing workflow has null inside.

**Comment by sfmskywalker** [comment]
Yeah that could be an issue. Do we have an established way of dealing with updating indexes? For example, is it customary to update indexes from the migration class? Do we bother at all?

**Pull request merged**